### PR TITLE
some Cargo Changes

### PR DESCRIPTION
**Pull request by jokoho48**
Fix findNearestVehicle dont return BOOL
Fix that findNearestVehicle dont return any time the nearest car
Small Performance Improvement in getSizeItem/handleDestroyed/startUnload
Enhance Some legibility

### CODE REVIEW

File: addons/cargo/functions/fnc_findNearestVehicle.sqf [hunk 1]
```diff
@@ -6,26 +6,26 @@
...
-if (_unit distance _vehicle <= MAX_LOAD_DISTANCE) exitwith {_vehicle};
+_loadTank = nearestObject [_unit, "tank"];
+if (_unit distance _loadTank <= MAX_LOAD_DISTANCE) exitwith {_loadTank};
 
-_loadtank = nearestObject [_unit, "ship"];
-if (_unit distance _vehicle <= MAX_LOAD_DISTANCE) exitwith {_vehicle};
+_loadShip = nearestObject [_unit, "ship"];
+if (_unit distance _loadShip <= MAX_LOAD_DISTANCE) exitwith {_loadShip};
 
 objNull;
```
Comment (jokoho48) [inline addons/cargo/functions/fnc_findNearestVehicle.sqf]: Todo: test if this is faster
```
private "_vehicles";
params ["_unit"];

_vehicles = nearestObjects [_unit , ["car", "air", "tank", "ship"], MAX_LOAD_DISTANCE]
if !(_vehicles params ["_vehicle"]) then{
    _vehicle
} else {
    objNull
};
```

Comment (jokoho48) [inline addons/cargo/functions/fnc_findNearestVehicle.sqf]: WorstCase is the same time BestCase Slower

File: addons/cargo/functions/fnc_getSizeItem.sqf [hunk 2]
```diff
@@ -6,15 +6,16 @@
...
  *
  * Public: No
  */
 
 #include "script_component.hpp"
 
-params ["_item"];
-if (isNumber (configFile >> "CfgVehicles" >> typeof _item >> QGVAR(size))) exitwith {
-    _item getvariable [QGVAR(size), getNumber (configFile >> "CfgVehicles" >> typeof _item >> QGVAR(size))];
+params ["_item", "_config"];
```
Comment (Glowbal) [inline addons/cargo/functions/fnc_getSizeItem.sqf]: Where does the `"_config"` come from?

Comment (jokoho48) [inline addons/cargo/functions/fnc_getSizeItem.sqf]: woops sry wrong line :/

File: addons/cargo/functions/fnc_startUnload.sqf [hunk 3]
```diff
@@ -13,19 +13,20 @@
...
-private ["_display", "_loaded", "_ctrl", "_selected", "_item"];
+private ["_display", "_loaded", "_loadedCount", "_ctrl", "_selected", "_item"];
 disableSerialization;
 _display = uiNamespace getvariable QGVAR(menuDisplay);
 if (isnil "_display") exitwith {};
 
 _loaded = GVAR(interactionVehicle) getvariable [QGVAR(loaded), []];
-if (count _loaded == 0) exitwith {};
+_loadedCount = count _loaded;
+if (_loadedCount == 0) exitwith {};
```
Comment (Glowbal) [inline addons/cargo/functions/fnc_startUnload.sqf]: I don't think storing the count in a variable is worth it

File: addons/cargo/functions/fnc_unloadItem.sqf [hunk 4]
```diff
@@ -52,7 +52,7 @@ _vehicle setvariable [QGVAR(loaded), _loaded, true];
 
 _space = [_vehicle] call FUNC(getCargoSpaceLeft);
 _itemSize = [_item] call FUNC(getSizeItem);
-_vehicle setvariable [QGVAR(space), _space + _itemSize, true];
+_vehicle setvariable [QGVAR(space), (_space + _itemSize), true];
```
Comment (Glowbal) [inline addons/cargo/functions/fnc_unloadItem.sqf]: Useless parenthesis.

File: addons/cargo/functions/fnc_startUnload.sqf (hunk 3 above)
Comment (jokoho48) [inline addons/cargo/functions/fnc_startUnload.sqf]: you use count 3 time and this reduce it to one(and you dont change something at Loaded)

File: addons/cargo/functions/fnc_unloadItem.sqf (hunk 4 above)
Comment (jokoho48) [inline addons/cargo/functions/fnc_unloadItem.sqf]: i think that improve the readebilty

**Pull request merged**